docs/manual/java/guide/cluster/PersistentEntityCassandra.md

```diff
@@ -22,4 +22,46 @@ In sbt:
 
 ## Configuration
 
+Lagom uses several internal Cassandra tables to persist entity data. These are known as the journal, which stores events, and the snapshot store, which stores snapshots of the state as an optimization for faster recovery (see [[Snapshots|PersistentEntity#Snapshots]] for details). [[Cassandra Read-Side support|ReadSideCassandra]] also uses a table to store the event offsets last processed by each read-side processor (detailed in [[Read-side design|ReadSide#Read-side-design]]). You will need to configure the keyspaces that contain the tables for each of these components.
+
+A keyspace in Cassandra is a namespace that defines data replication on nodes. Each service should use a unique keyspace name so that the tables of different services do not conflict with each other.
+
+You can configure these keyspaces in each service implementation project's `application.conf` file:
+
+```conf
+cassandra-journal.keyspace = users_journal
+cassandra-snapshot-store.keyspace = users_snapshot
+lagom.persistence.read-side.cassandra.keyspace = users_read_side
+```
+
+Cassandra keyspace names must start with an alphanumeric character and contain only alphanumeric and underscore characters. They are case-insensitive and stored in lowercase.
+
+While different services should be isolated by using different keyspaces, it is perfectly fine to use the same keyspace for all of these components within one service. In that case, it can be convenient to define a custom keyspace configuration property and use [property substitution](https://github.com/typesafehub/config#factor-out-common-values) to avoid repeating it.
+
+```conf
+users.cassandra.keyspace = users
+
+cassandra-journal.keyspace = ${users.cassandra.keyspace}
+cassandra-snapshot-store.keyspace = ${users.cassandra.keyspace}
+lagom.persistence.read-side.cassandra.keyspace = ${users.cassandra.keyspace}
+```
+
+By default, Lagom automatically creates the configured keyspaces and its internal tables if they are missing. If you prefer to manage the schema explicitly, you can disable automatic creation with these properties:
```

Comment:
I'm struggling a bit, to be honest. You say "this seemed like a new thought"... it kind of _is_ a new thought.

I can reorganize it like this:

-----

Lagom uses several internal Cassandra tables to persist entity data. These are known as the journal, which stores events, and the snapshot store, which stores snapshots of the state as an optimization for faster recovery (see [[Snapshots|PersistentEntity#Snapshots]] for details). [[Cassandra Read-Side support|ReadSideCassandra]] also uses a table to store the event offsets last processed by each read-side processor (detailed in [[Read-side design|ReadSide#Read-side-design]]). You will need to configure the keyspaces that contain the tables for each of these components.

A keyspace in Cassandra is a namespace that defines data replication on nodes. Each service should use a unique keyspace name so that the tables of different services do not conflict with each other.

Cassandra keyspace names must start with an alphanumeric character and contain only alphanumeric and underscore characters. They are case-insensitive and stored in lowercase.

By default, Lagom automatically creates the configured keyspaces and its internal tables if they are missing.

You can configure these keyspaces in each service implementation project's `application.conf` file:

```conf
cassandra-journal.keyspace = users_journal
cassandra-snapshot-store.keyspace = users_snapshot
lagom.persistence.read-side.cassandra.keyspace = users_read_side
```

While different services should be isolated by using different keyspaces, it is perfectly fine to use the same keyspace for all of these components within one service. In that case, it can be convenient to define a custom keyspace configuration property and use [property substitution](https://github.com/typesafehub/config#factor-out-common-values) to avoid repeating it.

```conf
users.cassandra.keyspace = users

cassandra-journal.keyspace = ${users.cassandra.keyspace}
cassandra-snapshot-store.keyspace = ${users.cassandra.keyspace}
lagom.persistence.read-side.cassandra.keyspace = ${users.cassandra.keyspace}
```

If you prefer to disable automatic creation of the keyspaces and tables, you can disable it with these properties:

```conf
cassandra-journal {
  keyspace-autocreate = false
  tables-autocreate = false
}
cassandra-snapshot-store {
  keyspace-autocreate = false
  tables-autocreate = false
}
lagom.persistence.read-side.cassandra {
  keyspace-autocreate = false
}
```

It is not possible to disable automatic creation of the offset store table at this time.

-----

To me, this now feels like the topic of automatic creation is now split into two disconnected parts that don't fit in well with their surrounding context. I'm happy to keep it like that if you think it's an improvement, though.